### ISSUE TITLE
Ansible lint do not validate some tasks that uses modules without collection path in name

### ISSUE THREAD

**Issue by czqrny** [ansible/ansible-lint#1835]
##### Summary

When I was testing my own rules I noticed that `ansible-lint` do not validate all tasks in my test playbooks. The problem is that `matchtask` method from `ansiblelint.rules.AnsibleLintRule` class do not even get such tasks for checking. 

This tasks are:

* `include_tasks`,
* `import_tasks`,
* probably more similar, but for example `include_vars` no

However, when I define this modules by full path (`ansible.builtin.include_tasks` and `ansible.builtin.import_tasks`), `ansible-lint` starts to see them. 

##### Issue Type

- Bug Report

##### Ansible and Ansible Lint details

```console (paste below)
ansible --version
ansible 2.9.27
  config file = None
  configured module search path = ['/home/tester/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/local/lib/python3.10/site-packages/ansible
  executable location = /usr/local/bin/ansible
  python version = 3.10.0 (default, Nov 17 2021, 15:26:39) [GCC 10.2.1 20210110]

ansible-lint --version
ansible-lint 5.3.3.dev13 using ansible 2.9.27
```

- ansible installation method: one of source, pip, OS package - `pip`
- ansible-lint installation method: one of source, pip, OS package - `pip`

##### OS / ENVIRONMENT

`python:3.10.0-bullseye` Docker container

##### STEPS TO REPRODUCE

Run ansible-lint command on such playbook, only tasks from second play reports `All tasks should be named` rule violation
 
```yaml
---
- name: "debug first playbook tasks not seen by ansible-lint"
  hosts: "localhost"
  tasks:
    - include_tasks: "debug1.yml"
      loop: [1, 2, 3, 4]
      loop_control:
        loop_var: myvar

    - import_tasks: "debug1.yml"

- name: "debug first playbook tasks can be seen by ansible=lint"
  hosts: "localhost"
  tasks:
    - ansible.builtin.include_tasks: "debug1.yml"
      loop: [1, 2, 3, 4]
      loop_control:
        loop_var: myvar

    - ansible.builtin.import_tasks: "debug1.yml"
 ```

```console (paste below)

```

##### Desired Behaviour

1. All tasks should be check and report that they have no name.
1. Maybe some other rule should report that full module path must be used.

##### Actual Behaviour

Only tasks from second play are reported as breaking rule.

Output of `ansible-lint` command:

```console
WARNING  Overriding detected file kind 'yaml' with 'playbook' for given positional argument: debug.yml
WARNING  Listing 2 violation(s) that are fatal
unnamed-task: All tasks should be named
debug.yml:15 Task/Handler: include_tasks  debug1.yml

unnamed-task: All tasks should be named
debug.yml:20 Task/Handler: import_tasks  debug1.yml

You can skip specific rules or tags by adding them to your configuration file:
# .ansible-lint
warn_list:  # or 'skip_list' to silence them completely
  - unnamed-task  # All tasks should be named
Finished with 2 failure(s), 0 warning(s) on 2 files.
```

**Comment by ssbarnea** [comment]
Working with current versions.
